lib/device.h

```diff
@@ -49,6 +49,7 @@ struct sync_device {
 	SOCKET sock;
 #endif
 	struct sync_io_cb io_cb;
+    const char *(*path_encode_cb)(const char *path);
```

Comment:
device.h isn't part of the "public" api, so I don't think it's enough to expose this here. We'd need some setter as well, I think...

Reply:
Roger, will fix that

Reply:
Also worth noticing that the indentation seems off here.